torch/nn/modules/rnn.py

```diff
@@ -848,6 +877,18 @@ def forward(self, input, hx=None):  # noqa: F811
             max_batch_size = int(max_batch_size)
         else:
             batch_sizes = None
+            is_batched = input.dim() == 3
+            batch_dim = 0 if self.batch_first else 1
+            if not is_batched:
+                input = input.unsqueeze(batch_dim)
+                if hx is not None:
+                    assert hx.dim() == 2, \
```

Comment:
Similarly here and below, a `RuntimeError` should be thrown instead of asserting